app/models/office.rb

```diff
@@ -1,6 +1,5 @@
 class Office < ApplicationRecord
-  has_many :providers
-  has_many :legal_aid_applications
-
   belongs_to :firm
+  has_many :legal_aid_applications
+  has_and_belongs_to_many :providers
```

Comment:
It's a long time since I've used `has_and_belongs_to_many`. Using a through relationship is so much more flexible, that I've always found that HABTM eventually get changed to using a through model. However, if it works for now fine - it's easy enough to change it later.

Reply:
Thinking about this a little further, one thing a through model will allow is the addition of a `firm` filter to a provider's offices. The HABTM would allow a providers to be associated with offices that do not belong to the provider's firm. Obviously there are other ways of restricting that, but it would be nice if the relationship handled that.

I'm not sure exactly what the provider <-> firm relationship is. I think that we have discussed providers moving firms, so that historically applications can exist for a particular provider that span multiple firms. There may even be instances where providers work for more that one firm at the same time.

Having a through model that also had an association with firm would allow for queries like `provider.offices.where(:provider_offices: { firm: firm })` which would be more complicated with a HABTM relationship.

Reply:
A firm has many offices.  A provider belongs to a firm and has a subset of that firm's offices, so I think HABTm works ok.

Reply:
Have discussed this with Julien and am happy for the association to be HABTM. We can change it when we need to.